### DISCUSSION TITLE
Combobox: Accepting object as well as value in ComboboxItem

### DISCUSSION THREAD

**Discussion by PixeledCode** [ariakit/ariakit#3508] [Questions] [answered]
Thanks for the awesome library.
Usually, Combobox components receive the `{value, label}[]` type. Currently, `ComboboxItem` only accepts `string` as a value, which is not ideal, as it should also support an object and when selection changes `ComboboxProvider` can provide the array of object in `setSelectedValue`. This allows a more flexible use case with the component.

Currently, I am doing `JSON.stringify` in ComboboxItem and then `JSON.parse` in setSelectedValue

**Comment by diegohaz** [comment, marked as answer]
Instead of `JSON.stringify`, you can maintain a separate array with your objects (which you probably already have) and retrieve the object for the value on `setSelectedValue`:

```tsx
const options = [
  { value: "a", label: "A" },
  { value: "b", label: "B" },
  { value: "c", label: "C" },
];

function getOptionFromValue(value: string) {
  return options.find((option) => option.value === value);
}

<ComboboxProvider
  setSelectedValue={(value) => {
    const option = getOptionFromValue(value);
    ...
  }}
```

**Reply by PixeledCode** [reply]
yeah that's what I did after realising JSON.stringify way will not work and is getting complicated
